README.md

```diff
@@ -37,50 +44,108 @@ expBackoffBase = 2
 maxRetries = UNLIMITED_RETRIES;
 ```
 
-### Running the example
+## SPIFFE SVID based KeyStore and TrustStore Provider
+
+### Install the SPIFFE Provider JAR
+
+Generate the JAR: 
 
-Required:
-* Linux (the example was tested in Linux Mint 18.3)
-* Java 8 
-* A running Spire server and agent. [HowTo](https://github.com/spiffe/spire#installing-spire-server-and-agent)
-* An entry in the registry that matches this application:
 ```
-cmd/spire-server/spire-server entry create \
--spiffeID spiffe://example.org/workload \
--parentID spiffe://example.org/host \
--selector unix:uid:1000
+./gradlew build
 ```
 
-To start the Java server you can run `./gradlew runExample`
+For installing the JAR file containing the provider classes as a bundled extension in the java platform, copy 
+`build/libs/spiffe-provider-0.1.0.jar` to `<java-home>/jre/lib/ext`
+
+### Configure `java.security` 
+
+Java Security Providers are configured in the master security properties file `<java-home>/lib/security/java.security`. 
+
+The way to register a provider is to specify the Provider subclass name and priority in the format
 
 ```
-$ ./gradlew runExample
+security.provider.<n>=<className>
+```
+
+This declares a provider, and specifies its preference order n.
 
-> Task :run
-[main] INFO spiffe.api.svid.examples.ClientExample - Fetching the SVIDs asynchronously
-[main] INFO spiffe.api.svid.examples.ClientExample - Waiting for certificates...
-[main] INFO spiffe.api.svid.examples.ClientExample - Doing other work...
-[main] INFO spiffe.api.svid.examples.ClientExample - Doing other work...
-[main] INFO spiffe.api.svid.examples.ClientExample - Doing other work...
-[grpc-default-executor-0] INFO spiffe.api.svid.examples.ClientExample - Spiffe ID fetched: spiffe://example.org/workload
-[main] INFO spiffe.api.svid.examples.ClientExample - Exiting...
+#### Register the SPIFFE Provider
 
-BUILD SUCCESSFUL in 1s
+You can extend and override the master security properties file. 
+
+Create a file `java.security` with the following content:
 
 ```
+security.provider.<n>=spiffe.provider.SpiffeProvider
 
-### Generating the JAR  
+# Determines the default key and trust manager factory algorithms for
+# the javax.net.ssl package.
+ssl.KeyManagerFactory.algorithm=Spiffe
+ssl.TrustManagerFactory.algorithm=Spiffe
 
+# The spiffeID that will be trusted
+ssl.spiffe.accept=spiffe://example.org/workload
```

Comment:
It would be great if this could also be configured directly in consuming code. For instance, if I have a service with multiple APIs, and I want some SPIFFE IDs to be accepted on one but not another. Not sure how this would normally be done in Java, but it's definitely something I think people will need.

Reply:
I think granting access to different URIs (APIs) based on the SPIFFE ID should be done in an authorization filter in the Tomcat container or in the Application, it cannot be part of the code of this Provider.